test/net/sourceforge/kolmafia/textui/RuntimeLibraryTest.java

```diff
@@ -1384,12 +1397,57 @@ void trackedByIncludesAllTracks() {
             actual,
             equalTo(
                 """
-            Returned: aggregate string [3]
-            0 => Transcendent Olfaction
-            1 => Gallapagosian Mating Call
-            2 => Red-Nosed Snapper
-            """));
+                                Returned: aggregate string [3]
+                                0 => Transcendent Olfaction
+                                1 => Gallapagosian Mating Call
+                                2 => Red-Nosed Snapper
+                                """));
       }
     }
   }
+
+  /**
+   * This test is intended to test running the maximizer from a command or script (not the GUI),
+   * show that the maximizer chooses a weapon and then emits the commands to equip that weapon.
+   */
+  @Test
+  public void itShouldMaximizeAndEquipSelectedWeapon() {
+    String maxStr = "effective";
+    HttpClientWrapper.setupFakeClient();
+    var cleanups =
+        new Cleanups(
+            withStats(10, 5, 5),
+            withEquippableItem("seal-skull helmet"),
+            withEquippableItem("astral shirt"),
+            withEquippableItem("old sweatpants"),
+            withEquippableItem("sewer snake"),
+            withEquippableItem("seal-clubbing club"));
+    String out;
+    String cmd;
+    cmd = "maximize(\"" + maxStr + "\", false)";
+    try (cleanups) {
+      out = execute(cmd);
+    }
+    assertFalse(out.isEmpty());
+    assertTrue(out.contains("Putting on seal-skull helmet..."));
+    assertTrue(out.contains("Wielding seal-clubbing club..."));
+    assertTrue(out.contains("Putting on old sweatpants..."));
+    assertContinueState();
+    var requests = getRequests();
+    assertTrue(requests.size() > 1);
+    int whichReq = -1;
+    for (int i = 0; i < requests.size(); i++) {
+      String body = getPostRequestBody(requests.get(i));
+      if (body.contains("whichitem=1")) {
+        whichReq = i;
+        break;
+      }
+    }
```

Comment:
probably this wants streaming, something like `requests.stream().filter(x -> getPostRequestBody(x).contains("whichitem=1")).findAny()`

Reply:
OK